### PR TITLE
fixed parsing of scenario matches

### PR DESCRIPTION
**Pull request by Skazu**
Created dedicated source branch for the issue mentioned here:
https://github.com/happyleavesaoc/aoc-mgz/pull/100
because i have additional fixes for another problem.

I'm closing the old PR.

Open in this PR:
Fixes for the Full-Body-Parser.

### CODE REVIEW

**Comment by Skazu** [comment]
FYI: I've added code to parse triggers/conditions for scenario records but for DE only, it doesn't break the old behaviour for aoc and others, and of course it's still in the fast header parser only.
There's also an example record commited.

I don't currently know if I will port the code into the full header parser, it would cost me quite a lot of time, the declarative approach and the long test duration makes it very difficult and unpleasant to debug there.
But I would still like to let you profit from our adaptations upstream. Maybe someone else who uses the full parser has the time to port the code?

Let me know if you have any suggestens for the code.

**Comment by happyleavesaoc** [comment]
It's fine to support only the fast header. As you say, anyone who wants can port it back given this example.

But, please make the tests pass (I guess by excluding the scenario file from the full tests).

**Comment by karmenat** [comment]
I have made a few code corrections and fixed the tests. From my point of view, the code is now ready-to-merge.

**Comment by Skazu** [comment]
I have made a few code corrections and fixed the tests. From my point of view, the code is now ready-to-merge.

File: tests/test_files.py [hunk 1]
```diff
@@ -32,5 +31,50 @@ def test_files_full(self):
         parse_file_full('tests/recs/de-13.07.aoe2record')
 
     def test_files_fast(self):
-        for path in glob.glob('tests/recs/*'):
```
Comment (happyleavesaoc) [inline tests/test_files.py:35]: Let's do this:

```
skip = {"tests/recs/de-50.6-scenario.aoe2record", ...}
for path in glob.glob('tests/recs/*'):
    if path in skip:
        continue
    parse_file_fast(path)
```

That way we don't need to update this test every time we add another test file.

**Comment by Skazu** [comment]
I've changed it.

**Comment by happyleavesaoc** [comment]
Thanks!

**Pull request merged**